plugins/terraform/terraform.go

```diff
@@ -0,0 +1,20 @@
+package terraform
+
+import (
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/needsauth"
+	"github.com/1Password/shell-plugins/sdk/schema"
+)
+
+func TerraformCLI() schema.Executable {
+	return schema.Executable{
+		Name:    "Terraform CLI",
+		Runs:    []string{"terraform"},
+		DocsURL: sdk.URL("https://developer.hashicorp.com/terraform/cli"),
+		NeedsAuth: needsauth.IfAll(
+			needsauth.NotForHelpOrVersion(),
+			needsauth.NotWithoutArgs(),
+		),
+		AllowedCredentials: schema.DynamicNumber,
```

Comment:
This PR should also the specification of the terraform API token, within this plugin.

Reply:
https://github.com/1Password/shell-plugins/issues/260